docs/topic-guides/blendable-web-apps/localization.md

```diff
@@ -0,0 +1,115 @@
+# Localizing Starcounter Apps
+
+Real world applications often need to support a variety of languages. This article describes how to add such support to your application. It doesn't describe any practices for translation of texts or internals of how localization support works in Starcounter.
+
+## Getting started
+
+1. Create resources file with strings you want to localize.
+   1. In Visual Studio, open "Add new item" menu and choose "Resources File"
+   1. Select any name for your file. We'll use `Strings.resx` in this example
+   1. Add as many resources files as many languages you want to support. Use the following format when naming them: `name.ll-cc.resx` where `ll` is the language code and `cc` is the country code. To continue our example, you would add `Strings.sv-se.resx` to support Swedish language as used in Sweden.
+1. Populate your resources files with texts you want to localize.
+   1. The `name` column should contain a valid C# indentifier, like `SignIn`. Value should be the relevant text in language chosen for this file. For example, it could be "Logga in" in file `Strings.sv-se.resx`. The comments column can be used freely.
+1. Visual Studio will generate a class from your resources file. In our example, it will generate `Strings` class with a string member `SignIn`
+   1. Value of this member will depend on currently selected culture. You don't have to worry about setting it in your application, it's done by a special language selection application.
+   1. In your application, instead of using hard-coded strings use members of this class. For example, here is an example of view-model using a generated resources class:
+
+   ```c#
+   public partial class SignInViewModel: Json
+   {
+       public string Message {get; set;}
+       public string Username {get; set;}
+       public string Password {get; set;}
+
+       public void Handle(Input.SignInTrigger input)
+       {
+           if(!_signInService.SignIn(Username, Password))
+           {
+               Message = Strings.WrongUsernameOrPassword;
+               return;
+           }
+           // ...
+       }
+   }
+   ```
+
+## Using placeholders
+
+Often you need to put some data in the texts you are localizing. For example you want to localize the text that includes the current user's name: 
+```c#
+Message = $"Hello, {user.FirstName}";
+```
+
+To do this, include placeholders (`{0}`, `{1}`, etc.) in your localized texts. Continuing our example, you would put the following value in `Strings.resx` for name `HelloUser`: `Hello, {0}`. To use it, use `string.Format` method:
+
+```c#
+Message = string.Format(Strings.HelloUser, user.FirstName);
```

Comment:
because `HelloUser` translation can put the user placeholder anywhere. E.g. `Hello, {0}!`. When using the string you shouldn't reason about its structure